### ISSUE TITLE
Allow empty jobParams

### ISSUE THREAD

**Issue by sbliven** [SciCatProject/backend#1411]
jobParams is currently required and also required to be non-empty. This should be optional for job types that don't require a `datasetIds` list.

Example: POST to /api/v3/jobs
```
{
  "type": "test",
  "jobParams": {},
  "ownerUser": "user1",
  "ownerGroup": "group1",
  "contactEmail": "a@b.com"
}
```

Gives error 
```
[Nest] 418  - 09/04/2024, 6:52:46 PM   ERROR {"status":400,"message":"Job parameters need to be defined."}
[Nest] 418  - 09/04/2024, 6:52:46 PM   ERROR HttpException: Job parameters need to be defined.
[Nest] 418  - 09/04/2024, 6:52:46 PM   ERROR Object:
{
  "requestUrl": "/api/v3/jobs",
  "requestUser": {
    "_id": "66cf29d54611a38cf2c100ba",
    "username": "admin",
    "email": "admin@scicat.project",
    "currentGroups": [
      "admin",
      "globalaccess",
      "ess"
    ],
    "authStrategy": "local"
  },
  "statusCode": 400
}
```

**Comment by sofyalaski** [comment]
I first wanted to fix it but then wasn't sure. This only throws an error if there are truly no jobParams. If you would put something like ProposalIds with an empty list, it creates a job ( doesn't do any checks though). I was just thinking that if there are no parameters - there is no job content.

**Comment by sbliven** [comment]
Not necessarily. For instance, if you had a `test` job that just had a LogAction configured then it wouldn't need any jobParams but could still be useful for checking network connectivity.

**Comment by sofyalaski** [comment]
So you think we should accept both no jobParams and empty ones? Or as Despina proposed, no jobParams is accepted but if they are provided they cannot be empty?

**Comment by sofyalaski** [comment]
addressed in PR #1433. 
If you have no jobParams you must pass an empty object. JobParams is a required property in DTO. 
Empty object as jobParams is legit now.
